util-core/src/main/scala/com/twitter/concurrent/AsyncStream.scala

```diff
@@ -609,6 +609,7 @@ object AsyncStream {
     extends AsyncStream[A] {
     private[this] lazy val _more: AsyncStream[A] = next()
     def more(): AsyncStream[A] = _more
+    override def toString(): String = s"Cons($fa, $next)"
```

Comment:
cool. also please don't hesitate to change it in internal review.

`$next` isn't so useful here -- it's always going to be `"<function0>"`. something much more useful is if we made a `Lazy[A]` type. i had a ticket for that idea, you might be able to dig it up. for the `Lazy[A]` type, `toString` would return`"<function0>"` if not yet evaluated, otherwise it would show the evaluated `A`.